src/Schema/Element/Element.php

```diff
@@ -1,60 +1,140 @@
 <?php
 namespace GoetasWebservices\XML\XSDReader\Schema\Element;
 
+use DOMElement;
+use DOMXPath;
 use GoetasWebservices\XML\XSDReader\Schema\Item;
+use GoetasWebservices\XML\XSDReader\Schema\Schema;
+use GoetasWebservices\XML\XSDReader\SchemaReader;
+use GoetasWebservices\XML\XSDReader\SchemaReaderLoadAbstraction;
 
 class Element extends Item implements ElementItem, ElementSingle
 {
-
+    /**
+    * @var int
+    */
     protected $min = 1;
 
+    /**
+    * @var int
+    */
     protected $max = 1;
 
+    /**
+    * @var bool
+    */
     protected $qualified = false;
 
+    /**
+    * @var bool
+    */
     protected $nil = false;
 
+    /**
+    * @return int
+    */
     public function getMin()
     {
         return $this->min;
     }
 
+    /**
+    * @param int $min
+    *
+    * @return $this
+    */
     public function setMin($min)
     {
         $this->min = $min;
         return $this;
     }
 
+    /**
+    * @return int
+    */
     public function getMax()
     {
         return $this->max;
     }
 
+    /**
+    * @param int $max
+    *
+    * @return $this
+    */
     public function setMax($max)
     {
         $this->max = $max;
         return $this;
     }
 
+    /**
+    * @return bool
+    */
     public function isQualified()
     {
         return $this->qualified;
     }
 
+    /**
+    * @param bool $qualified
+    *
+    * @return $this
+    */
     public function setQualified($qualified)
     {
-        $this->qualified = (boolean) $qualified;
+        $this->qualified = is_bool($qualified) ? $qualified : (boolean) $qualified;
         return $this;
     }
 
+    /**
+    * @return bool
+    */
     public function isNil()
     {
         return $this->nil;
     }
 
+    /**
+    * @param bool $nil
+    *
+    * @return $this
+    */
     public function setNil($nil)
     {
-        $this->nil = (boolean) $nil;
+        $this->nil = is_bool($nil) ? $nil : (boolean) $nil;
```

Comment:
what is the difference with the previous version?

Reply:
as with the (boolean), it's regarding phpstan / psalm extra-paranoid type safety.